### ISSUE TITLE
Latest and default LaunchTemplate label fails to create nodes

### ISSUE THREAD

**Issue by mcavoyk** [awslabs/karpenter#434]
Using the provisioner label `node.k8s.aws/launch-template-version` set to `$Default` or `$Latest` causes Karpenter to attempt to create Kubernetes nodes with these labels.

```bash
2021-05-27T20:47:39.785Z        ERROR   Continuing after failing to bind, creating node ip-172-16-198-124.ec2.internal, 
Node "ip-172-16-198-124.ec2.internal" is invalid: metadata.labels: Invalid value: "$Latest": a valid label must be 
an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric 
character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```

### Environment
Kubernetes Version: v1.17.12-eks-7684af
Karrpenter Version: v0.2.5

**Comment by ellistarn** [comment]
Very interesting. This appears to be a incompatibility between AWS naming and Kubernetes name requirements. IMO, we probably should simply default to $Default if nothing is provided. This would mean that we either don't support $Latest or we build another mechanism for it and $Default. This could be as simple as substituting something for the '$' (e.g. alt36) to keep compatibility with kubernetes label naming.

I'm tentatively in support of simply not supporting "$Latest". Given what I know of docker image tagging, latest is something of an antipattern. If we do this then default would simply be the unspecified case (we do this currently https://github.com/awslabs/karpenter/blob/404b9e36618d4d8d9a4e08461d85a3b634f1fd58/pkg/cloudprovider/aws/constraints.go#L28) . Any thoughts on this given your use case?

@JacobGabrielson are you up for taking this on?

**Comment by ellistarn** [comment]
Hey @mcavoyk, I'm curious why you need a custom launch template at all. It's a bit tricky, since you need to make sure you include cluster.resourcesVpcConfig.securityGroupIds in the launch template's security group, or they can't connect to the masters. What feature are you looking for?

**Comment by JacobGabrielson** [comment]
Sorry for the delay in getting to this. It seems like the base problem here is trying to create a `label` on a Node that has invalid characters (from the launch template version). I'm wondering if instead of being saved as a label, the launch-template-version should be stored as an annotation? I'm not sure this really needs to be stored as a label in the first place (eg not sure one really wants to do queries based on launch-template-version)?

**Comment by ellistarn** [comment]
>I'm wondering if instead of being saved as a label, the launch-template-version should be stored as an annotation?
Labels are critical to how scheduling works, since pods are only able to select nodes based on their labels (not annotations). For a pod to declare a launch template scheduling constraint, it *must* happen through labels.

**Comment by JacobGabrielson** [comment]
I'm not sure launch template version would fall into that category? For one thing, it doesn't make sense to select a node based on launch-template-version *alone*. And even then I'm struggling to see a real world case. For example, it would have to be:

Pod A wants to schedule onto launch-template-name=foo and launch-template-version=4, Pod B wants to schedule onto launch-template-name=foo and launch-template-version=3. I could construct a very artificial scenario where this would happen but I can't think of a realistic one. (But maybe I'm missing something?)

**Comment by ellistarn** [comment]
Another option here would be to simply not support launch template version and *always* use $Default.

**Comment by akestner** [comment]
@JacobGabrielson I understand you had been working on this, have you been able to make an additional progress? It'd be great to get this into the v0.3 release

**Comment by JacobGabrielson** [comment]
Fixed by #595
